pkg/registries/registries_test.go

```diff
@@ -39,6 +39,45 @@ func TestScopeMatchesRegistry(t *testing.T) {
 	}
 }
 
+func TestIsSubScopeOfReg(t *testing.T) {
+	for _, tt := range []struct {
+		scope, reg string
+		expected   bool
+	}{
+		{"foo.example.com", "*.example.com", true},          // Valid sub-scope match
+		{"*.foo.example.com", "*.example.com", true},        // Valid sub-scope match
+		{"foo.example.com/bar", "*.example.com", true},      // Valid sub-scope match
+		{"foo.registry.com", "*.example.com", false},        // Invalid sub-scope match
+		{"foo.example.com", "**.example.com", false},        // Invalid sub-scope entry
+		{"foo.example.com", "example.*.com", false},         // Invalid sub-scope entry
+		{"foo.example.com", "*.example.com/foo/bar", false}, // Invalid wildcard entry
+	} {
+		t.Run(fmt.Sprintf("%#v, %#v", tt.scope, tt.reg), func(t *testing.T) {
+			res := isSubScopeOfReg(tt.scope, tt.reg)
+			assert.Equal(t, tt.expected, res)
+		})
+	}
+}
+
+func TestIsValidWildcard(t *testing.T) {
+	for _, tt := range []struct {
+		scope    string
+		expected bool
+	}{
+		{"*.example.com", true},             // Valid wildcard
+		{"**.example.com", false},           // Invalid wildcard entry
+		{"example.*.com", false},            // Invalid wildcard entry
+		{"*example.com/foo/bar", false},     // Invalid wildcard entry
+		{"*example.com:foo", false},         // Invalid wildcard entry
+		{"*example.com/foo:sha@bar", false}, // Invalid wildcard entry
```

Comment:
These should probably start with `*.` so that they actually exercise the matching of `/:@` instead of just the `*` rule.

Reply:
Fixed